tests/tests_pytorch/models/test_tpu.py

```diff
@@ -366,20 +324,23 @@ def test_if_test_works_with_checkpoint_false(tmpdir):
     assert trainer.state.finished, f"Training failed with {trainer.state}"
 
 
-@RunIf(tpu=True)
-def test_tpu_sync_dist():
-    """Test tpu spawn sync dist operation."""
+def xla_launch(function):
+    strategy = TPUSpawnStrategy(parallel_devices=list(range(8)))
+    launcher = _XLALauncher(strategy=strategy)
+    return launcher.launch(function, strategy)
```

Comment:
@carmocca each "function" here must probably call `.setup_environment()` before you can call any distributed collectives. You are seeing the error about `Accessing the XLA device before processes have spawned is not allowed.` because the test is trying to access strategy.device before `.setup_environment()` is called. 

In Trainer, this is done right after processes get spawned, and in Lite this is done in the function that wraps around the run() method.

Reply:
Why would the launcher not take care of this?

Reply:
I think you are getting confused. 
I can see that you called setup_environment on the main process with the trainer changes. This calls torch.cuda.set_device on the main process, which is not correct. The error message is well justified in this case. 

I would definitely call setup_environment in your special test manually. Otherwise you just end up implementing https://github.com/Lightning-AI/lightning/pull/11073